zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/ControllerStrictJsonCompareTest.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest;
+
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.junit.AnalyzeClasses;
+import com.tngtech.archunit.junit.ArchTest;
+import com.tngtech.archunit.lang.ArchRule;
+import com.tngtech.archunit.lang.syntax.ArchRuleDefinition;
+import org.springframework.test.web.reactive.server.WebTestClient.BodyContentSpec;
+
+@AnalyzeClasses(
+    packages = "io.camunda.zeebe.gateway.rest",
+    importOptions = ImportOption.OnlyIncludeTests.class)
+public class ControllerStrictJsonCompareTest {
+
+  /** This ArchUnit test ensures that any REST API controller tests use JsonCompareMode.STRICT */
+  @ArchTest
+  public static final ArchRule RULE_USE_STRICT_JSON_COMPARISON =
+      ArchRuleDefinition.noClasses()
+          .should()
+          .callMethod(BodyContentSpec.class, "json", String.class);
```

Comment:
❓ so this ensures `#json` is never called with just passing in just json string, which implicitly means LENIENT, right? This still allows us to explicitly use LENIENT when we want to. IS there some way we could also limit it explicitly only to passing in `JsonCompareMode.STRICT`?

Reply:
> ❓ so this ensures #json is never called with just passing in just json string, which implicitly means LENIENT, right?

Yes, exactly 👍 

> This still allows us to explicitly use LENIENT when we want to. IS there some way we could also limit it explicitly only to passing in JsonCompareMode.STRICT?

There probably is but I wouldn't want to be that restrictive. If we have a good argument for going for LENIENT, let's use that. But that's a conscious decision than that we can question.